conftest.py

```diff
@@ -3,8 +3,54 @@
 from enum import Enum
 import requests
 from typing import Callable
+import logging
+from dataclasses import dataclass
 
+log = logging.getLogger(__name__)
 
+@dataclass
+class SensorInfo:
+    name: str
+    hid: str
+    model: str
+    firmware_version: int
+    reading_interval: int
+
+    def __post_init__(self):
+        if not isinstance(self.name, str):
+            raise TypeError("'name' should be a string")
+        
+        if self.name == "":
+            raise ValueError("'name' should not be empty")
+        
+
+        if not isinstance(self.hid, str):
+            raise TypeError("'hid' should be a string")
+
+        if self.hid == "":
+            raise ValueError("'hid' should not be empty")
+        
+
+        if not isinstance(self.model, str):
+            raise TypeError("'model' should be a string")
+
+        if self.model == "":
+            raise ValueError("'model' should not be empty")
+
+
+        if not isinstance(self.firmware_version, int):
+            raise TypeError("'firmware_version' should be an integer")
+
+        if self.firmware_version == "":
+            raise ValueError("'firmware_version' should not be empty")   
+
+        
+        if not isinstance(self.reading_interval, int):
+            raise TypeError("'reading_interval' should be an integer")
+
+        if self.reading_interval == "":
```

Comment:
Same thing as in my first comment applies here. We're interested in checking that this is a valid value from sensor's standpoint. That would be seeing that it's equal or more than 1.